test/test-mongocrypt-marking.c

```diff
@@ -1344,21 +1345,105 @@ static void validate_text_search_ciphertext(_mongocrypt_tester_t *tester,
         size_t tscount = 0;
         ASSERT(bson_iter_init_find(&b_iter, &b_bson, "s"));
         tscount = validate_text_search_token_set_array_common(&b_iter, crypt);
-        ASSERT((tscount > 0) == !!(flags & TEXTSEARCH_SPEC_HAS_SUBSTRING));
+        ASSERT(expected_tag_counts.substrings == tscount);
```

Comment:
nit - 
```suggestion
        ASSERT_CMPSIZE_t(expected_tag_counts.substrings, ==, tscount);
```

for all of these numerical comparison assertions, so that we get more debugging info on failure.